src/CMakeLists.txt

```diff
@@ -88,6 +94,8 @@ if(WIN32 AND EMBED_SSH_PATH)
 	target_sources(git2internal PRIVATE ${SRC_SSH})
 
 	list(APPEND LIBGIT2_SYSTEM_INCLUDES "${EMBED_SSH_PATH}/include")
+        list(APPEND LIBGIT2_SYSTEM_LIBS "crypt32" "bcrypt")
+        list(APPEND LIBGIT2_PC_LIBS "-lcrypt32" "-lbcrypt")
 	file(WRITE "${EMBED_SSH_PATH}/src/libssh2_config.h" "#define HAVE_WINCNG\n#define LIBSSH2_WINCNG\n#include \"../win32/libssh2_config.h\"")
```

Comment:
In fact, instead of trying to create a libssh2 config, why don't we just treat it as a normal dep and run `cmake` on it, exactly like we do for our other dependencies?